### DISCUSSION TITLE
aspect ratio distribution of 2d grains

### DISCUSSION THREAD

**Discussion by liang-tech** [idaholab/moose#25417] [Q&A General]
Hello. Does MOOSE have the function to calculate aspect ratio distribution of 2d grains?

In the phase-field simulation, we can get a final grain structure. Then we focus on the geometric properties of grains that are related to ellipses fitted to the grains and calculate aspect ratio distribution like the pictures.

![EllipseBasedParameters_02](https://github.com/idaholab/moose/assets/81635815/4d722953-ad7f-4ac0-b1ec-35da9644912d)
![搜狗截图20230907100045](https://github.com/idaholab/moose/assets/81635815/28bba6bc-6033-4e8b-af58-8cb0346798bd)

**Comment by lindsayad** [comment]
@dschwen
